pkg/types/aws/validation/machinepool.go

```diff
@@ -52,6 +50,40 @@ func ValidateMachinePool(platform *aws.Platform, p *aws.MachinePool, fldPath *fi
 	return allErrs
 }
 
+func validateVolumeSize(p *aws.MachinePool, fldPath *field.Path) field.ErrorList {
```

Comment:
Nit: this function is so small/simple that it could maybe be integrated into the caller `if`.